pkg/sharing/secret_import_reconciler.go

```diff
@@ -92,6 +94,42 @@ func (r *SecretImportReconciler) AttachWatches(controller controller.Controller)
 			return result
 		},
 	})
+	if err != nil {
+		return err
+	}
+
+	// Watch namespaces partly so that we cache them because we migh be doing a lot of lookups
+	return controller.Watch(&source.Kind{Type: &corev1.Namespace{}}, &enqueueNamespaceToSecret{
+		ToRequests: r.mapNamespaceToSecret,
+		Log:        r.log,
+	})
+}
+
+func (r *SecretImportReconciler) mapNamespaceToSecret(ns client.Object) []reconcile.Request {
+	var secretList corev1.SecretList
+	err := r.client.List(context.Background(), &secretList, client.InNamespace(ns.GetName()))
+	if err != nil {
+		// TODO what should we really do here?
```

Comment:
Can we remove this comment or provide more details?

Reply:
i can remove it but i think the detail is : "I don't know what to do here."   I copied over the comment from existing code setting up a similar watch -- someone else was unsure what to do here also. see:
https://github.com/vmware-tanzu/carvel-secretgen-controller/blob/39cc31078f2ea13daa7aaa662d9c91308de85864/pkg/sharing/secret_import_reconciler.go#L74

and
https://github.com/vmware-tanzu/carvel-secretgen-controller/blob/8fc61ebfb2fed4a8d7bfac2c193cafb3149dafd7/pkg/sharing/secret_reconciler.go#L67

I'm not saying it's the _most_ helpful comment,  but i think leaving it in shows that the original author didn't have informed intentions --  like it's not, "oh it must be this way for a reason". it's just "oh it's this way because we had to _something_ on the error, and `panic` seemed like the wrong answer."